bitnami/redis/README.md

```diff
@@ -526,6 +526,40 @@ tls-client-cert-file
 tls-ca-cert-file
 ```
 
+A custom Lua script can be added to the `redis-exporter` by way of the `metrics.extraArgs.script` parameter.  The pathname of the script *MUST* exist on the container, or the `redis_exporter` process (and therefore the whole pod) will refuse to start.  The script can be provided to the sidecar containers via the `extraVolumes` and `extraVolumeMounts` parameters:
+
+```
+my-redis-deploy:
+  master: &volumeConfigs
+    extraVolumeMounts:
+      - name: '{{ printf "%s-metrics-script-file" (include "common.names.fullname" .) }}'
+        mountPath: '{{ printf "/mnt/%s/" (include "common.names.name" .) }}'
+        readOnly: true
+    extraVolumes:
+      - name: '{{ printf "%s-metrics-script-file" (include "common.names.fullname" .) }}'
+        configMap:
+          name: '{{ printf "%s-metrics-script" (include "common.names.fullname" .) }}'
+  replica:
+    <<: *volumeConfigs
+```
+
+Then deploy the script into the correct location via `extraDeploy`:
+
+```
+my-redis-deploy:
+  metrics:
+    extraArgs:
+      script: "/mnt/my-redis-deploy/my_custom_metrics.lua"
+  extraDeploy:
+    - apiVersion: v1
+      kind: ConfigMap
+      metadata:
+        name: '{{`{{ printf "%s-metrics-script" (include "common.names.fullname" .) }}`}}'
+      data:
+        my_custom_metrics.lua: |
+          -- a super-clever Lua script goes here
+```
```

Comment:
1. Configuring/disabling TLS in the metrics sidecar is arguably also a niche activity, and it's documented, precisely because it's somewhat non-obvious.
2. It really only works if the chart is configured in a very specific way, i.e., the `extraVolumeMounts.mountPath` and `extraVolumeMounts.name` need to match, and that really needs to be spelled out *somewhere* so that users can figure it out, since the responsibility of "getting it right" is implicitly pushed onto them.
3. More documentation can only be a good thing.

Reply:
I totally agree that it's not trivial and we can make an effort to have better docs. That said, the right place to document this day-2 advanced topics is our docs system. I mean, the site below:

- https://docs.bitnami.com/kubernetes/infrastructure/redis/

And.. The good thing about that doc system is that is also open-source, see https://github.com/bitnami/charts-docs. It would be just a matter of creating a new guide in the folder below:

- https://github.com/bitnami/charts-docs/tree/main/charts/redis